cartography/intel/aws/emr.py

```diff
@@ -48,53 +57,76 @@ def get_emr_describe_cluster(boto3_session: boto3.session.Session, region: str,
     return cluster_details
 
 
+@dataclass
+class EMRClusterNodeProperties(CartographyNodeProperties):
+    arn: PropertyRef = PropertyRef('ClusterArn')
+    auto_terminate: PropertyRef = PropertyRef('AutoTerminate')
+    autoscaling_role: PropertyRef = PropertyRef('AutoScalingRole')
+    custom_ami_id: PropertyRef = PropertyRef('CustomAmiId')
+    firstseen: PropertyRef = PropertyRef('firstseen')
+    id: PropertyRef = PropertyRef('Id')
+    instance_collection_type: PropertyRef = PropertyRef('InstanceCollectionType')
+    lastupdated: PropertyRef = PropertyRef('lastupdated', set_in_kwargs=True)
+    log_encryption_kms_key_id: PropertyRef = PropertyRef('LogEncryptionKmsKeyId')
+    log_uri: PropertyRef = PropertyRef('LogUri')
+    master_public_dns_name: PropertyRef = PropertyRef('MasterPublicDnsName')
+    name: PropertyRef = PropertyRef('Name')
+    outpost_arn: PropertyRef = PropertyRef('OutpostArn')
+    region: PropertyRef = PropertyRef('Region', set_in_kwargs=True)
+    release_label: PropertyRef = PropertyRef('ReleaseLabel')
+    repo_upgrade_on_boot: PropertyRef = PropertyRef('RepoUpgradeOnBoot')
+    requested_ami_version: PropertyRef = PropertyRef('RequestedAmiVersion')
+    running_ami_version: PropertyRef = PropertyRef('RunningAmiVersion')
+    scale_down_behavior: PropertyRef = PropertyRef('ScaleDownBehavior')
+    security_configuration: PropertyRef = PropertyRef('SecurityConfiguration')
+    servicerole: PropertyRef = PropertyRef('ServiceRole')
+    termination_protected: PropertyRef = PropertyRef('TerminationProtected')
+    visible_to_all_users: PropertyRef = PropertyRef('VisibleToAllUsers')
+
+
+@dataclass
+class EMRClusterToAwsAccountRelProperties(CartographyRelProperties):
+    lastupdated: PropertyRef = PropertyRef('lastupdated', set_in_kwargs=True)
+
+
+@dataclass
+# (:EMRCluster)<-[:RESOURCE]-(:AWSAccount)
+class EMRClusterToAWSAccount(CartographyRelSchema):
+    target_node_label: str = 'AWSAccount'
+    target_node_key: str = 'id'
+    target_node_key_property_ref: PropertyRef = PropertyRef('AccountId', set_in_kwargs=True)
+    direction: LinkDirection = LinkDirection.INWARD
+    rel_label: str = "RESOURCE"
+    properties: EMRClusterToAwsAccountRelProperties = EMRClusterToAwsAccountRelProperties()
+
+
+@dataclass
+class EMRClusterSchema(CartographyNodeSchema):
+    label: str = 'EMRCluster'
+    properties: EMRClusterNodeProperties = EMRClusterNodeProperties()
+    sub_resource_relationship: CartographyRelSchema = EMRClusterToAWSAccount()
+
+
 @timeit
 def load_emr_clusters(
-    neo4j_session: neo4j.Session, cluster_data: List[Dict], region: str, current_aws_account_id: str,
-    aws_update_tag: int,
+        neo4j_session: neo4j.Session,
+        cluster_data: List[Dict],
+        region: str,
+        current_aws_account_id: str,
+        aws_update_tag: int,
 ) -> None:
-    query = """
```

Comment:
This PR refactors the EMR sync to demonstrate using the new functionality. Rather than hand write queries, module authors now define schema objects and call `build_ingestion_query()` and `load_graph_data()`.